packages/dev/core/src/Materials/Node/Blocks/PBR/iridescenceBlock.ts

```diff
@@ -0,0 +1,184 @@
+        );
+    }
+
+    /**
+     * Initialize the block and prepare the context for build
+     * @param state defines the state that will be used for the build
+     */
+    public initialize(state: NodeMaterialBuildState) {
+        state._excludeVariableName("iridescenceOut");
+        state._excludeVariableName("vIridescenceParams");
+    }
+
+    /**
+     * Gets the current class name
+     * @returns the class name
+     */
+    public getClassName() {
+        return "IridescenceBlock";
+    }
+
+    /**
+     * Gets the intensity input component
+     */
+    public get intensity(): NodeMaterialConnectionPoint {
+        return this._inputs[0];
+    }
+
+    /**
+     * Gets the indexOfRefraction input component
+     */
+    public get indexOfRefraction(): NodeMaterialConnectionPoint {
+        return this._inputs[1];
+    }
+
+    /**
+     * Gets the thickness input component
+     */
+    public get thickness(): NodeMaterialConnectionPoint {
+        return this._inputs[2];
+    }
+
+    /**
+     * Gets the iridescence object output component
+     */
+    public get iridescence(): NodeMaterialConnectionPoint {
+        return this._outputs[0];
+    }
+
+    public autoConfigure() {
+        if (!this.intensity.isConnected) {
+            const intensityInput = new InputBlock("Iridescence intensity", NodeMaterialBlockTargets.Fragment, NodeMaterialBlockConnectionPointTypes.Float);
+            intensityInput.value = 1;
+            intensityInput.output.connectTo(this.intensity);
+
+            const indexOfRefractionInput = new InputBlock("Iridescence ior", NodeMaterialBlockTargets.Fragment, NodeMaterialBlockConnectionPointTypes.Float);
+            indexOfRefractionInput.value = 1.3;
+            indexOfRefractionInput.output.connectTo(this.indexOfRefraction);
+
+            const thicknessInput = new InputBlock("Iridescence thickness", NodeMaterialBlockTargets.Fragment, NodeMaterialBlockConnectionPointTypes.Float);
+            thicknessInput.value = 400;
+            thicknessInput.output.connectTo(this.thickness);
+        }
+    }
+
+    public prepareDefines(mesh: AbstractMesh, nodeMaterial: NodeMaterial, defines: NodeMaterialDefines) {
+        super.prepareDefines(mesh, nodeMaterial, defines);
+
+        defines.setValue("IRIDESCENCE", true, true);
+        defines.setValue("IRIDESCENCE_TEXTURE", false, true);
+        defines.setValue("IRIDESCENCE_THICKNESS_TEXTURE", false, true);
+    }
+
+    public bind(effect: Effect, nodeMaterial: NodeMaterial, mesh?: Mesh) {
+        super.bind(effect, nodeMaterial, mesh);
+
+        const intensity = this.intensity.connectInputBlock?.value ?? 1;
+        const indexOfRefraction = this.indexOfRefraction.connectInputBlock?.value ?? PBRIridescenceConfiguration._DefaultIndexOfRefraction;
+        const thickness = this.thickness.connectInputBlock?.value ?? PBRIridescenceConfiguration._DefaultMaximumThickness;

```

Comment:
I think it's too restrictive to assume the thickness is coming from an input block: it means the thickness will be a constant and can't come from a texture (for eg).

Maybe you can follow what was done for the thickness input of the `SubSurface` block?

Same thing for the intensity.

Relevant code in SubSurfaceBlock:

https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Materials/Node/Blocks/PBR/subSurfaceBlock.ts#L143-L168

Reply:
Yup good catch, still NME n00b :-)